### ISSUE TITLE
Inclusion of ember-cli app into another ember-cli app

### ISSUE THREAD

**Issue by maitriyogin** [broccolijs/broccoli#193]
Hi,
I'm trying to embed another ember-cli app into a parent ember-cli app, to achieve some kind of shell application that can include and display multiple child apps.
I've heard that because of the es6 name spacing I can do this without having to use conductor.js or oasis.js.
I'm stuck with trying to include he child projects and feeling that I'm missing some kind of broccoli definition to get the child into the build?
I've bower installed the child and have tried ember cli's app.import('childapp') which actually adds the child apps files to the build. Although when trying to import anything from the child's namespace it just doesn't work...
I feel that I'm missing something fundamental, any guidance would be most appreciated.
Cheers,
Stephen

**Comment by joliss** [comment]
I'll go ahead and close this. Having child builds work seamlessly is pretty hard to do in general, for fundamental reasons related to npm - probably not worth tackling at the moment.

**Comment by stefanpenner** [comment]
Ultimately, the goal here is no-different then what we do with ember-addons today, essentially pluggable parts of the broccoli build pipeline. Broccoli supports our use-case sufficiently here.

TL;DR put your app in an add-on, and you should be good to go from a build tooling perspective. From an app perspective their exist many complexities. I suspect fleshing out these parts will be part of the ember 2.x push

**Comment by thermokarst** [comment]
@stefanpenner: revisiting this topic one year later, do you think this situation can be viewed in a different light now? Or, do you suspect that addressing the app-perspective complexities are largely unaffected by changes in the ember/broccoli landscape?

**Comment by stefanpenner** [comment]
We can already nest broccoli builds. In fact all addons are this. And glimmer is included in ember like this.

I guess, I'm not sure really what the question is or how it relates to broccoli itself.

**Comment by thermokarst** [comment]
Thanks for following up! I wanted to get your take on nested builds (I want to reuse an ember app in a few different scenarios). I don't think my question is specifically related to broccoli --- I just didn't want to open a new issue when this thread already existed. I can certainly move this over somewhere else (e.g. ember-cli) if that would be more appropriate. Thanks!

**Comment by thermokarst** [comment]
It looks like ember engines will potentially handle my needs, whenever it lands. Thanks!

**Comment by fsmanuel** [comment]
@thermokarst you can use the [addon](https://github.com/dgeb/ember-engines)
